modules/vaos/app/controllers/vaos/preferences_controller.rb

```diff
@@ -2,15 +2,29 @@
 
 module VAOS
   class PreferencesController < VAOS::BaseController
-    def index
-      response = preferences_service.get_preferences(current_user)
+    def show
+      response = preferences_service.get_preferences
+      render json: VAOS::PreferencesSerializer.new(response)
+    end
+
+    def update
+      response = preferences_service.put_preferences(put_params)
       render json: VAOS::PreferencesSerializer.new(response)
     end
 
     private
 
     def preferences_service
-      VAOS::PreferencesService.new
+      VAOS::PreferencesService.new(current_user)
+    end
+
+    def put_params
+      params.require(:notification_frequency)
```

Comment:
Is notification frequency the only required param?

Reply:
Yes. Good question. I did a little bit of reverse engineering / testing of the endpoint on staging in the console and it looks that way. If the FE identifies other validations and required fields they can always let us know and we'll add those.

Reply:
You can chain like this: `params.require(:params).permit(:other, :params)`

Reply:
Not a big fan of chaining these since the behavior is considerably different when you pass an array to require. If it were more consistent in what require returns, I might consider it.